ballista/rust/core/src/execution_plans/shuffle_writer.rs

```diff
@@ -244,12 +260,26 @@ impl ShuffleWriterExec {
                                             stream.schema().as_ref(),
                                         )?;
 
+                                        let cur_bytes = writer.num_bytes;
                                         writer.write(&output_batch)?;
+                                        total_bytes_count.fetch_add((writer.num_bytes - cur_bytes) as usize, Ordering::Acquire);
```

Comment:
nit: The `writer` here is new so we can just assume `cur_bytes` is 0

```suggestion
                                        writer.write(&output_batch)?;
                                        total_bytes_count.fetch_add(writer.num_bytes  as usize, Ordering::Acquire);
```